data/mods/ssb/abilities.ts

```diff
@@ -196,887 +472,1759 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
+			},
+			onTryMovePriority: -2,
+			onTryMove(pokemon, target, move) {
+				if (!pokemon.hasAbility('ic3peak')) {
+					pokemon.removeVolatile('ic3peak');
+					return;
+				}
+				if (this.effectData.lastMove === move.id && pokemon.moveLastTurnResult) {
+					this.effectData.numConsecutive++;
+				} else {
+					this.effectData.numConsecutive = 0;
+				}
+				this.effectData.lastMove = move.id;
+			},
+			onBasePowerPriority: 24,
+			onBasePower(basePower, pokemon, target, move) {
+				if (move.refrigerateBoosted) return;
+				const dmgMod = [1, 2, 3, 4, 5];
+				const numConsecutive = this.effectData.numConsecutive > 4 ? 4 : this.effectData.numConsecutive;
+				console.log(dmgMod[numConsecutive]);
+				return this.chainModify(dmgMod[numConsecutive]);
+			},
+		},
 		isNonstandard: "Custom",
+		gen: 8,
 	},
+
 	// GMars
-	mysteryshell: {
-		desc: "If this Pokemon is a Minior in its Meteor forme, it cannot be afflicted by a status condition. This Pokemon cannot be hit with a critical hit.",
-		shortDesc: "Status immunity while in Meteor forme, crit immunity.",
-		name: "Mystery Shell",
-		isNonstandard: "Custom",
+	capsulearmor: {
+		desc: "While in Minior-Meteor forme, this Pokemon cannot be affected by major status conditions and is immune to critical hits. This ability cannot be ignored by Moongeist Beam, Sunsteel Strike, Mold Breaker, Teravolt, or Turboblaze.",
+		shortDesc: "Minior-Meteor: Immune to crits and status",
+		name: "Capsule Armor",
+		isPermanent: true,
+		isUnbreakable: true,
 		onCriticalHit: false,
 		onSetStatus(status, target, source, effect) {
 			if (target.species.id !== 'miniormeteor' || target.transformed) return;
 			if ((effect as Move)?.status) {
-				this.add('-immune', target, '[from] ability: Mystery Shell');
+				this.add('-immune', target, '[from] ability: Capsule Armor');
 			}
 			return false;
 		},
+		onTryAddVolatile(status, target) {
+			if (target.species.id !== 'miniormeteor' || target.transformed) return;
+			if (status.id !== 'yawn') return;
+			this.add('-immune', target, '[from] ability: Capsule Armor');
+			return null;
+		},
 	},
-	// guishark
-	gzguishark: {
-		desc: "Boosts Attack by one stage upon switch-in or Mega Evolution.",
-		shortDesc: "Boosts Attack by 1 stage upon switch-in/Mega Evolution.",
-		name: "gz guishark",
-		isNonstandard: "Custom",
-		onStart(pokemon) {
-			this.boost({atk: 1}, pokemon);
+
+	// grimAuxiliatrix
+	biosteel: {
+		desc: "This Pokemon restores 1/3 of its maximum HP, rounded down, when it switches out, and other Pokemon cannot lower this Pokemon's stat stages.",
+		shortDesc: "Regenerator + Clear Body.",
+		name: "Bio-steel",
+		onSwitchOut(pokemon) {
+			pokemon.heal(pokemon.baseMaxhp / 3);
 		},
+		onBoost(boost, target, source, effect) {
+			if (source && target === source) return;
+			let showMsg = false;
+			let i: BoostName;
+			for (i in boost) {
+				if (boost[i]! < 0) {
+					delete boost[i];
+					showMsg = true;
+				}
+			}
```

Comment:
```suggestion
			for (const i in boost) {
				if (boost[i]! < 0) {
					delete boost[i];
					showMsg = true;
				}
			}
```
Just do this for all your `for` loops please

Reply:
ditto